addons/survey/static/src/js/survey_form.js

```diff
@@ -728,6 +728,9 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
                         subQuestionsIds.forEach(function (id) {
                             if (!((questionId + '_' + id) in data)) {
                                 errors[questionId] = constrErrorMsg;
+                                // Highlight missing rows headers in matrix questions
+                                let row = self.el.querySelector("div[id='" + questionId + "']>.o_survey_question_matrix>tbody>tr[id='" + id + "']>th")
```

Comment:
Initially I only though about this (f-string counterpart and using `const` instead of `let` when the reference should not change).

```suggestion
                                const row = self.el.querySelector(`div[id="${questionId}"] > .o_survey_question_matrix > tbody > tr[id="${id}"] > th`);
```

Or even

```suggestion
                               self.el
                               .querySelector(`div[id="${questionId}"] > .o_survey_question_matrix > tbody > tr[id="${id}"] > th`)
                               .classList.add('bg-danger');
```
Then I saw that the question Id selector is reconstructed at every step which is not necessary, and for the same number of lines we can get a more efficient:
```js
const subQuestionsIds = $questionWrapper.find('table').data('subQuestions');
// Highlight unanswered rows' header
const questionBodySelector = `div[id="${questionId}"] > .o_survey_question_matrix > tbody`;
subQuestionsIds.forEach((subQuestionId) => {
    if (!(`${questionId}_${subQuestionId}` in data)) {
        errors[questionId] = constrErrorMsg;
        self.el.querySelector(`${questionBodySelector} > tr[id="${subQuestionId}"] > th`).classList.add('bg-danger');
    }
}
```
This also included some suggestions for wording, renaming of non-explicit `id`, spacing conventions in CSS, use of `const`, use of arrow functions syntax for shortness.
All because we are in master and I think we can afford a slightly bigger diff to improve a few things while we are here. Take what you like, I'm also available to give longer explanations as needed :)

Reply:
Hello @flch-odoo :slightly_smiling_face: 
Even more cleaner this way I agree!
Updated my code :slightly_smiling_face:
Thanks for your advice and the thorough follow-up as always !